### PR TITLE
Feature Update routers to handle registered user requests

### LINKED ISSUE #73: Create registered-user-endpoint for each calculator
**__Task Description__**
Create a registered-user-endpoint for each calculator and implement on calculators so it is used for registered users.
If session is present, calculator should query the database and save data from carbon to the user's profile.
Calculator should reroute to user dashboard at `/account` if session is present.

**__Additional Considerations__**
Any supplemental information including unresolved questions, links to external resources, screenshots, etc.

### PR DESCRIPTION
**Pull request by TiciaD**
- update Electricity router to update database when user session is present
  - update Vehicle router to update database when user session is present
  - update Fuel router to update database when user session is present
  - update Flight router to update database when user session is present
  - update prisma schema with new fields and updated old field names to better match Carbon API response
  - added mock-server missing response field

Closes #73

### CODE REVIEW

**Comment by TiciaD** [comment]
I know this PR has a lot of files changed but essentially it just allows us to save the Carbon API responses to the database if the user is signed in. So to test just make sure you login, fill in a calculator and check prisma studio to make sure the data was saved using `npx prisma studio`.

File: src/server/router/electricity.router.ts [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
+            data: {
+              userId: user.id,
+              electricity_value: carbonResponse.electricity_value,
+              electricity_unit: carbonResponse.electricity_unit,
+              estimated_at: carbonResponse.estimated_at,
+              carbon_g: carbonResponse.carbon_g,
+            },
+          })
+          .then((response) => response.userId);
+        updateUser;
```
Comment (ialej001) [inline src/server/router/electricity.router.ts]: Does this line do anything?

File: src/server/router/flight.router.ts [hunk 2]
```diff
@@ -1,22 +1,55 @@
...
+            estimated_at: carbonResponse.estimated_at,
+            carbon_g: carbonResponse.carbon_g,
+            distance_value: carbonResponse.distance_value,
+            distance_unit: carbonResponse.distance_unit,
+            flightLeg: {
+              createMany: {
+                data: carbonResponse.legs.map(
+                  (leg: FlightLeg, index: number) => {
+                    return {
+                      departure_airport: leg.departure_airport,
```
Comment (ialej001) [inline src/server/router/flight.router.ts]: Can this be rewritten as 
`{...leg, legNumber: index+1}`?

File: src/server/router/fuel.router.ts [hunk 3]
```diff
@@ -1,18 +1,39 @@
...
       )
     );
 
-    return response.json().then((data) => {
+    let carbonResponse: FuelResponse = await response.json().then((data) => {
       return data.data.attributes;
     });
+
+    if (user) {
+      const updateUser = await ctx.prisma.fuelUsed
```
Comment (ialej001) [inline src/server/router/fuel.router.ts]: updateFuelUse instead of updateUser

File: src/server/router/electricity.router.ts (hunk 1 above)
Comment (TiciaD) [reply src/server/router/electricity.router.ts]: I think this was a remnant when I was testing to see if it worked, it was essentially a `console.log`, I removed it!

File: src/server/router/flight.router.ts (hunk 2 above)
Comment (TiciaD) [reply src/server/router/flight.router.ts]: Updated!

File: src/server/router/fuel.router.ts (hunk 3 above)
Comment (TiciaD) [reply src/server/router/fuel.router.ts]: Updated!

**Review by ialej001 — APPROVED** [review]
Excellent work!

**Review by victoriacheng15 — APPROVED** [review]

**Pull request merged**